libraries/fc/include/fc/static_variant.hpp

```diff
@@ -14,587 +14,86 @@
 #include <typeinfo>
 #include <fc/exception/exception.hpp>
 #include <boost/core/typeinfo.hpp>
+#include <variant>
 
 namespace fc {
 
-// Implementation details, the user should not import this:
-namespace impl {
-
-template<typename R, int N, typename... Ts>
-struct storage_ops;
-
-template<typename X, typename... Ts>
-struct position;
-
-template<int Pos, typename... Ts>
-struct type_at;
-
-template<typename... Ts>
-struct type_info;
-
-template< typename Visitor, typename ...Ts>
-struct const_result_type_info;
-
-template< typename Visitor, typename ...Ts>
-struct result_type_info;
-
-template< typename Visitor, typename ...Ts>
-struct rvalue_ref_result_type_info;
+// This will go away.
+template<typename Result>
+struct visitor {};
 
-template<typename StaticVariant>
-struct copy_construct
+template <typename variant, int I = 0>
+void from_index(variant& v, int index) 
 {
-   StaticVariant& sv;
-   copy_construct( StaticVariant& s ):sv(s){}
-   template<typename T>
-   void operator()( const T& v )const
-   {
-      sv.init(v);
-   }
-};
-
-template<typename StaticVariant>
-struct move_construct
+  if constexpr(I >= std::variant_size_v<variant>)
+  {
+    throw std::runtime_error{"Variant index " + std::to_string(I + index) + " out of bounds"};
```

Comment:
should be std::exception or FC::exception? and should catch all std exception here and rethrow with fc::exception?

Reply:
Which exception type is preferred?

Reply:
fc

Reply:
yeah, exiting static_variant throws FC exception, and the current error handling logic is based on handling FC exception, so need to keep using FC.